### DISCUSSION TITLE
Support Workload Identity / IRSA

### DISCUSSION THREAD

**Discussion by jkroepke** [grafana/grafana-infinity-datasource#768] [Ideas]
Currently infinity datasource does not support any cloud native authentication related to Azure.

This isn't an issue, because the Authentication with Client Credentials is fully OAUTH2 compatible. I can use the OAUTH2 Auth backend.

However, we would like to avoid any static credentials in future. The Grafana builtin datasource for AWS Cloudwatch and Azure Monitor both support "SDK" authentication which support Workload Identity out of the box.

I would like to use infinity-datasource for the MS Graph API to fetch details around our M365 tenant.

**Comment by tobiasehlert** [comment]
Would be pretty awesome if we could get this to work, so that it's possible to use an UAMI to access remote things without requiring the use of an App Registration.

@jkroepke, why was the #785 not finished? Looked like it was kind of ready to me.

**Reply by jkroepke** [reply]
Generally, the feedback and communication loop with maintains was not satisfied, including in other PRs as well. At some point, we just dropped alls our upstream efforts and implement a custom solution in our environment to break the chain.
